### PR TITLE
sstable: update TestWriter to exercise TableFormatPebblev5

### PR DESCRIPTION
**Pull request by jbowens**
Update the TestWriter test to exercise the new TableFormatPebblev5 that enables
columnar blocks. This commit also fixes a couple issues with key ordering checks.

### CODE REVIEW

**Review by RaduBerinde — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/pebble/3973#-:-O835fSV0tYTmJuQ_IWW:bnfp4nl)* status: 0 of 18 files reviewed, 1 unresolved discussion (waiting on @jbowens)

___
*[`sstable/writer_test.go` line 42 at r5](https://reviewable.io/reviews/cockroachdb/pebble/3973#-O835NV584MMeUncvrNo:-O835NV584MMeUncvrNp:b-eanokh) ([raw file](https://github.com/cockroachdb/pebble/blob/c04151249b44cbc5c5182d03b822d93eb55f9fb7/sstable/writer_test.go#L42)):*
> ```Go
> 		TableFormatPebblev5: "testdata/writer_v5",
> 	}
> 	for format, file := range formatFiles {
> ```

[nit] It's kind of annoying for subtest order to be non-deterministic. Could make it an array and skip empty locations

**Review by jbowens — COMMENTED** [review]
TFTR!

Will wait until #3980 is merged to merge this.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/pebble/3973#-:-O83Xv2X47bN4WnZgENH:bk42cyz)* status: 0 of 18 files reviewed, all discussions resolved (waiting on @RaduBerinde)

___
*[`sstable/writer_test.go` line 42 at r5](https://reviewable.io/reviews/cockroachdb/pebble/3973#-O835NV584MMeUncvrNo:-O83XwS710fiqjZt5WDI:bb6un1j) ([raw file](https://github.com/cockroachdb/pebble/blob/c04151249b44cbc5c5182d03b822d93eb55f9fb7/sstable/writer_test.go#L42)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

[nit] It's kind of annoying for subtest order to be non-deterministic. Could make it an array and skip empty locations
</blockquote></details>

Done

**Pull request merged**
